djangochannelsrestframework/mixins.py

```diff
@@ -89,3 +92,66 @@ def delete(self, **kwargs):
 
     def perform_delete(self, instance, **kwargs):
         instance.delete()
+
+
+class PaginatedModelListMixin(ListModelMixin):
+
+    pagination_class = api_settings.DEFAULT_PAGINATION_CLASS
+
+    @action()
+    def list(self, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset(**kwargs), **kwargs)
+        page = self.paginate_queryset(queryset, **kwargs)
+        if page is not None:
+            serializer = self.get_serializer(
+                instance=page, many=True, action_kwargs=kwargs
+            )
+            return self.get_paginated_response(serializer.data), status.HTTP_200_OK
+
+        serializer = self.get_serializer(
+            instance=queryset, many=True, action_kwargs=kwargs
+        )
+        return serializer.data, status.HTTP_200_OK
+
+    @property
+    def paginator(self) -> Optional[any]:
+        """Gets the paginator class
+
+        Returns:
+            Pagination class. Optional.
+        """
+        if not hasattr(self, "_paginator"):
+            if self.pagination_class is None:
+                self._paginator = None
+            else:
+                self._paginator = self.pagination_class()
+        return self._paginator
+
+    def paginate_queryset(
+        self, queryset: QuerySet[Model], **kwargs: Dict
+    ) -> Optional[QuerySet]:
+        if self.paginator is None:
+            return None
+        return self.paginator.paginate_queryset(
+            queryset, self.scope, view=self, **kwargs
+        )
+
+    def get_paginated_response(
+        self, data: Union[ReturnDict, ReturnList]
+    ) -> OrderedDict:
+        assert self.paginator is not None
+        return self.paginator.get_paginated_response(data)
+
+
+class StreamedPaginatedListMixin(PaginatedModelListMixin):
+    async def handle_action(self, action: str, request_id: str, **kwargs):
+        await super().handle_action(action, request_id, **kwargs)
```

Comment:
@action decorated methods so not need to always return a tuple, this can also call the send method themselves. 


you can call the `self.reply` method


```python
self.reply(request_id=request_id, action=action, data=data, status=status)
```

so instead of overriding `handle_action`  i think it would be better to override the `list method`

```
async def list(self, request_id, action, **kwargs):
     data, status = await super().list(action=action, request_id=request_id, **kwargs)
     await self.reply(action=action, data=data, status=status, request_id=request_id)
     
     while self.paginator.offset < self.paginator.count:
         count = self.paginator.count
         limit = self.paginator.limit
         offset = self.paginator.offset
         kwargs["offset"] = limit + offset
         
         data, status = await super().list(action=action, request_id=request_id, **kwargs)
         await self.reply(action=action, data=data, status=status, request_id=request_id)
         
```

might be worthwhile using the limit, offset info in the response from the `super().list` method.